src/daft-functions/src/numeric/clip.rs

```diff
@@ -0,0 +1,99 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    datatypes::{DataType, InferDataType},
+    prelude::{Field, Schema},
+    series::Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct Clip;
+
+#[typetag::serde]
+impl ScalarUDF for Clip {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        "clip"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        if inputs.len() != 3 {
+            return Err(DaftError::SchemaMismatch(format!(
+                "Expected 3 input arguments (array, min, max), got {}",
+                inputs.len()
+            )));
+        }
+        let array_field = inputs[0].to_field(schema)?;
+        let min_field = inputs[1].to_field(schema)?;
+        let max_field = inputs[2].to_field(schema)?;
+
+        // Check if the array_field is numeric
+        if !array_field.dtype.is_numeric() {
+            return Err(DaftError::TypeError(format!(
+                "Expected array input to be numeric, got {}",
+                array_field.dtype
+            )));
+        }
+
+        // Check if min_field and max_field are numeric or null
+        if !(min_field.dtype.is_numeric() || min_field.dtype == DataType::Null) {
+            return Err(DaftError::TypeError(format!(
+                "Expected min input to be numeric or null, got {}",
+                min_field.dtype
+            )));
+        }
+
+        if !(max_field.dtype.is_numeric() || max_field.dtype == DataType::Null) {
+            return Err(DaftError::TypeError(format!(
+                "Expected max input to be numeric or null, got {}",
+                max_field.dtype
+            )));
+        }
```

Comment:
These should be consolidated in `InferDataType` instead.

Reply:
Moved the validation logic to `clip_op`, now it throws an error if any of these conditions are violated.